// @dotenv/cli/lib/commands/pull.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const command_1 = require("@oclif/command");
const append_to_gitignore_service_1 = require("../services/append-to-gitignore-service");
const warn_if_env_project_does_not_exist_service_1 = require("../services/warn-if-env-project-does-not-exist-service");
const pull_service_1 = require("../services/pull-service");
const deprecation_service_1 = require("../services/deprecation-service");
class Pull extends command_1.Command {
    async run() {
        const { argv, flags } = this.parse(Pull);
        new deprecation_service_1.DeprecationService().run('pull');
        // 1. create gitignore
        await new append_to_gitignore_service_1.AppendToGitignoreService().run();
        // 2. check if .env.project file exists
        await new warn_if_env_project_does_not_exist_service_1.WarnIfEnvProjectDoesNotExistService({ _this: this }).run();
        // 3. pull
        await new pull_service_1.PullService(argv[0], argv[1], flags.dotenv_me || '').run();
    }
}
exports.default = Pull;
Pull.description = 'pull .env';
Pull.args = [
    {
        name: 'environment',
        required: false,
        description: 'Pull .env.ci, .env.staging, and .env.production',
        hidden: false,
        default: 'development',
        options: ['development', 'ci', 'staging', 'production'],
    },
    {
        name: 'filename',
        required: false,
        description: 'Set output filename. Defaults to .env for development and .env.{environment} for other environments',
        hidden: false,
    },
];
Pull.flags = {
    dotenv_me: command_1.flags.string({
        char: 'm',
        description: 'pass value for .env.me rather than reading from .env.me file',
        hidden: false,
        multiple: false,
        env: 'DOTENV_ME',
        required: false,
    }),
};
