// @zoralabs/nft-drop-contracts/dist/typechain/factories/MetadataRenderAdminCheck__factory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetadataRenderAdminCheck__factory = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [],
        name: "Access_OnlyAdmin",
        type: "error",
    },
];
const _bytecode = "0x6080604052348015600f57600080fd5b50603f80601d6000396000f3fe6080604052600080fdfea2646970667358221220f95acdc9a7a07607174fd26b32178cad4a51baf898f98a790a5240981339ec7764736f6c634300080a0033";
const isSuperArgs = (xs) => xs.length > 1;
class MetadataRenderAdminCheck__factory extends ethers_1.ContractFactory {
    constructor(...args) {
        if (isSuperArgs(args)) {
            super(...args);
        }
        else {
            super(_abi, _bytecode, args[0]);
        }
    }
    deploy(overrides) {
        return super.deploy(overrides || {});
    }
    getDeployTransaction(overrides) {
        return super.getDeployTransaction(overrides || {});
    }
    attach(address) {
        return super.attach(address);
    }
    connect(signer) {
        return super.connect(signer);
    }
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.MetadataRenderAdminCheck__factory = MetadataRenderAdminCheck__factory;
MetadataRenderAdminCheck__factory.bytecode = _bytecode;
MetadataRenderAdminCheck__factory.abi = _abi;
